### PR TITLE
Remove GH link from resources nav section for Waypoint

### PR DESCRIPTION
**Pull request by heatlikeheatwave**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-heat-choreremove-github-link-wa-76488f-hashicorp.vercel.app/waypoint) 🔎
- [Asana task](https://app.asana.com/0/0/1206697393359304/f) 🎟️

## 🗒️ What

- Remove GitHub link from resources section of sidebar nav

## 🤷 Why

Part of Waypoint IA restructuring

## 🧪 Testing
1. Visit Waypoint landing [page](https://dev-portal-git-heat-choreremove-github-link-wa-76488f-hashicorp.vercel.app/waypoint). The GitHub link should not be in the "resources" section of the sidebar nav.
2. Visit Waypoint docs [page](https://dev-portal-git-heat-choreremove-github-link-wa-76488f-hashicorp.vercel.app/waypoint/docs). The GitHub link should not be in the "resources" section of the sidebar nav.
3. Visit specific docs [page](https://dev-portal-git-heat-choreremove-github-link-wa-76488f-hashicorp.vercel.app/waypoint/docs/runner/config). The GitHub link should not be in the "resources" section of the sidebar nav.

Screenshot 2024-02-29 at 17 55 03

### CODE REVIEW

File: src/components/sidebar/helpers/generate-resources-nav-items.ts [hunk 1]
```diff
@@ -150,12 +151,16 @@ function generateResourcesNavItems(
 			title: 'Support',
 			href: DEFAULT_SUPPORT_LINK,
 		},
-		{
-			title: 'GitHub',
-			href: productSlug
-				? GITHUB_LINKS_BY_PRODUCT_SLUG[productSlug]
-				: DEFAULT_GITHUB_LINK,
-		},
+		...(productSlug !== 'waypoint'
```
Comment (RubenSandwich) [inline src/components/sidebar/helpers/generate-resources-nav-items.ts:154]: Not a fan of this as I'd rather the data filtered out before it hits this logic, but we are using the same logic just a few lines up: https://github.com/hashicorp/dev-portal/blob/ed5bff48fa84a63dbbe162a8659a008913a40734/src/components/sidebar/helpers/generate-resources-nav-items.ts#L134 So I'll allow it.

**Review by RubenSandwich — APPROVED** [review]
Everything looks good and works. Thanks for the test plan. ❤️

File: src/components/sidebar/helpers/generate-resources-nav-items.ts (hunk 1 above)
Comment (heatlikeheatwave) [reply src/components/sidebar/helpers/generate-resources-nav-items.ts:154]: The default here would be the HashiCorp org. I asked in the ticket if Tony & Emily are okay with that, as that would be preferable here! Will let you know.

Comment (RubenSandwich) [reply src/components/sidebar/helpers/generate-resources-nav-items.ts:154]: 🤷🏻‍♂️

**Pull request merged**